tools/submission/submission-checker.py

```diff
@@ -428,34 +429,39 @@ def check_performance_dir(config, model, path):
-    if target_latency:
-        if int(rt['99.00 percentile latency (ns)']) > target_latency:
-            log.error("%s Latency constraint not met, expected=%s, found=%s",
-                         fname, target_latency, rt['99.00 percentile latency (ns)'])
-
-    # Check Minimum queries were issued to meet test duration
-    min_query_count = config.get_min_query_count(model, scenario)
-    if int(rt['min_query_count']) < min_query_count:
-        log.error("%s Required minimum Query Count not met by user config, Expected=%s, Found=%s",
-                      fname, min_query_count, rt['min_query_count'])
-    if scenario == "Offline" and (int(rt['samples_per_query']) < OFFLINE_MIN_SPQ):
-        log.error("%s Required minimum samples per query not met by user config, Expected=%s, Found=%s",
-                      fname, OFFLINE_MIN_SPQ, rt['samples_per_query'])
-
-    # Test duration of 60s is met
-    if int(rt["min_duration (ms)"]) < TEST_DURATION_MS:
-         log.error("%s Test duration lesser than 60s in user config. expected=%s, found=%s",
-                       fname, TEST_DURATION_MS, rt["min_duration (ms)"])
+    if config.version != "v0.5":
+        # not supported for v0.5
+
+        # check if the benchmark meets latency constraint
+        target_latency = config.latency_constraint.get(model, dict()).get(scenario)
+        if target_latency:
+            if int(rt['99.00 percentile latency (ns)']) > target_latency:
+                log.error("%s Latency constraint not met, expected=%s, found=%s",
+                            fname, target_latency, rt['99.00 percentile latency (ns)'])
+
+        # Check Minimum queries were issued to meet test duration
+        min_query_count = config.get_min_query_count(model, scenario)
+        if int(rt['min_query_count']) < min_query_count:
+            log.error("%s Required minimum Query Count not met by user config, Expected=%s, Found=%s",
+                        fname, min_query_count, rt['min_query_count'])
+        if scenario == "Offline" and (int(rt['samples_per_query']) < OFFLINE_MIN_SPQ):
+            log.error("%s Required minimum samples per query not met by user config, Expected=%s, Found=%s",
+                        fname, OFFLINE_MIN_SPQ, rt['samples_per_query'])
+
+        # Test duration of 60s is met
+        if int(rt["min_duration (ms)"]) < TEST_DURATION_MS:
+            log.error("%s Test duration lesser than 60s in user config. expected=%s, found=%s",
+                        fname, TEST_DURATION_MS, rt["min_duration (ms)"])
 
     return is_valid, res
 
 
-def files_diff(list1, list2):
+def files_diff(list1, list2, optional=None):
     """returns a list of files that are missing or added."""
+    if not optional:
+        optional = []
     if list1 and list2:
-        for i in ["mlperf_log_trace.json", "results.json"]:
+        for i in ["mlperf_log_trace.json", "results.json"] + optional:
```

Comment:
Why not fold `["mlperf_log_trace.json", "results.json"]` into `optional`?

Reply:
ok.

Reply:
done.